client/client_test.go

```diff
@@ -59,7 +59,6 @@ func (ss *notifyingSender) reset(waiter *sync.WaitGroup) {
 
 func (ss *notifyingSender) wait() {
 	ss.waiter.Wait()
```

Comment:
Actually, I think we should just remove `notifyingSender.wait` and call `wg.Wait` directly. This way `ss.waiter` isn't shared by both reader and writer and we can nil it out after calling `Done`.